apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c

```diff
@@ -223,19 +840,41 @@ void itc_arastorage_db_get_result_message_p(void)
 * @scenario             If used api returns DB_OK
 * @apicovered           db_print_header , db_query
 * @precondition         Database resource must be initialized
-* @postcondition        NA
+* @postcondition        none
 */
 void itc_arastorage_db_print_header_p(void)
 {
+	TC_ASSERT_EQ("ARASTORAGE_STARTUP", g_check, true);
 	db_result_t ret;
 
+	g_cursor = NULL;
 	snprintf(g_query, QUERY_LENGTH, "SELECT %s, %s, %s FROM %s WHERE %s > 5;", g_attribute_set[0], g_attribute_set[1], g_attribute_set[2], RELATION_NAME, g_attribute_set[0]);
 	g_cursor = db_query(g_query);
 	TC_ASSERT_NEQ("db_query", g_cursor, NULL);
 	ret = db_print_header(g_cursor);
 	TC_ASSERT_EQ_CLEANUP("db_print_header", DB_ERROR(ret), 0, db_cursor_free(g_cursor));
 	db_cursor_free(g_cursor);
 	g_cursor = NULL;
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_arastorage_db_print_header_n
+* @brief            Print a header of query result with invalid argument
+* @scenario         Print relation, attribute name of query result with NULL value
+* @apicovered       db_print_header
+* @precondition     none
+* @postcondition    none
+*/
+static void itc_arastorage_db_print_header_n(void)
+{
+	TC_ASSERT_EQ("ARASTORAGE_STARTUP", g_check, true);
```

Comment:
what is g_check??

Reply:
g_check is global flag used in "itc_arastorage_startup_p".  It is marked as false on arastorage_startup failure. Arastorage_startup needs to run successfully before running other ITCs.  Inside those ITCs , g_check validates the successful completion of Arastorage_startup function.